src/celeritas/optical/interactor/RayleighInteractor.hh

```diff
@@ -80,25 +80,19 @@ CELER_FUNCTION Interaction RayleighInteractor::operator()(Engine& rng) const
     do
     {
         new_dir = sample_direction(rng);
+        // Project polarization onto plane perpendicular to new direction.
+        // In the unlikely case that the incident polarization and new
+        // direction are coincident, make_orthogonal will return a zero vector,
+        // and all elements of new_pol will be NaN, which will reject the
+        // sample.
```

Comment:
Won't `nan < fmax * u` return false and accept the sample?

Reply:
Yep, "false" means "reject" rather than "accept" 🤪  I've allowed it and updated the rayleigh sampler.

Reply:
Does it make sense to always accept nan? I think rejecting it is reasonable, but I can't think of a case where we'd want to accept it as a sample.

Reply:
Accepting it "propagates" the value, which some of our algorithms (e.g. clamp/min/max do) do to make sure we don't hide errors. In contrast some functions such as fmax ignore nans (so `f(nan, x) -> x`). I've reverted the behavior so we throw on it.